docs/System_Metrics.md

```diff
@@ -0,0 +1,213 @@
+__average__ over the polling interval should be reported.
+
+- Some devices report a load average in addition to CPU utilization. This value should **not** be reported.
+
+- Some devices mislabel load average as CPU utilization and vice-versa. The value to be reported should be done 
+after careful inspection and determination of which is the actual CPU utilization.
+
+- The CPU should be classified as `control` or `data`.
+
+- CPUs should either report a `cpu_name` or `cpu_no` or both.
+
+#### Structure
+* metrics_group_type: cpu
+* Metrics: 
+    - cpu_utilization:\<GAUGE\>
+* Dimensions
+    - cpu_type:\<string\> (control, data, etc.)
+    - cpu_no:\<string\> and/or
+    - cpu_name:\<string\>
+              
+#### Example
+```
+cpu_utilization:56.2
+cpu_type:control
+cpu_no:5.1
+cpu_name:packet
+```
+
+## Memory
+
+Memory is always normalized to bytes.  'Total' memory should always be the amount of physical memory, and this should 
+be split where necessary into DRAM or Heap.
+
+- Total (physical memory) and utilized memory in bytes. Downstream systems would calculate percentage utilization.
+
+- Total (physical memory)/utilized would be reported for each type of memory - e.g. dram or heap.
+
+- If the memory metrics are reported by the device in kilobytes and megabytes (or some other unit), they should be 
+normalized to bytes.
+
+#### Structure
+* metrics_group_type: memory
+* Metrics: 
+    - memory_total:\<GAUGE\>
+    - memory_used:\<GAUGE\>
+* Dimensions
+    - memory_type:\<string\> (heap, dram, etc.)
+              
+#### Example
+```
+memory_total:4294967296
+memory_used:1932735283
+memory_type:dram
+```
+
+## Environment
+
+Environment is currently limited to Power Supplies and Fans.
+
+- Total number of fans and working fans.
+
+- Total number of power supplies and working power supplies.  Note that some devices will return a power supply as 
+'working' without it being energized.
+
+#### Structure
+* metrics_group_type: environment
+* Metrics: 
+    - fans_total:\<GAUGE\>
+    - fans_ok:\<GAUGE\>
+    - power_units_total:\<GAUGE\>
+    - power_units_on:\<GAUGE\>
+              
+#### Example
+```
+fans_total:4
+fans_ok:4
+power_units_total:2
+power_units_on:1
+```
+
+## Temperature
+
+Temperature should be normalized to Fahrenheit for each sensor available.
+
+#### Structure
+* metrics_group_type: temperature
+* Metrics: 
+    - temperature_fahrenheit:\<GAUGE\>
+* Dimensions:
+    - sensor:\<string\>
+              
+#### Example
+```
+temperature_fahrenheit:100.5
+sensor:cpu_5_1
+```
+
+## Status
+
+The possible values for `device status` are enumerated in [DEVICE_METRICS_STATES].  Effectively an integer is returned 
+according to the states.
+
+- (0) Success
+- (1) Authentication Failure
+- (2) Network Failure
+- (3) Timeout
+- (4) Partial Metric Failure
+- (5) Internal Failure
+- (6) Missing Metrics
+- (7) Ping Failure
+
+#### Structure
+* metrics_group_type: status
+* Metrics: 
+    - status:\<GAUGE\>
```

Comment:
The convention should be "polling_status" (though this can be configured through "polling_status_metric_name" -- for example, we use "functional_polling_status" as the metric name in the Functional metrics plugins.)